lib/EnsEMBL/REST/Model/ga4gh/ga4gh_utils.pm

```diff
@@ -0,0 +1,119 @@
+=head1 LICENSE
+
+Copyright [1999-2015] Wellcome Trust Sanger Institute and the EMBL-European Bioinformatics Institute
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+     http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+
+=cut
+
+# shared configuration etc
+package EnsEMBL::REST::Model::ga4gh::ga4gh_utils;
+
+
+use Moose;
+extends 'Catalyst::Model';
+
+use Bio::EnsEMBL::IO::Parser::VCF4Tabix;
+use Bio::EnsEMBL::Variation::DBSQL::VCFCollectionAdaptor;
+use Digest::MD5 qw(md5_hex);
+with 'Catalyst::Component::InstancePerContext';
+use Data::Dumper;
+has 'context' => (is => 'ro');
+
+
+sub build_per_context_instance {
+  my ($self, $c, @args) = @_;
+  return $self->new({ context => $c, %$self, @args });
+}
+
+
+sub fetch_VCFcollection_by_id{
+
+  my $self         = shift;
+  my $variantSetId = shift; ## may be null
+  $ENV{ENSEMBL_VARIATION_VCF_ROOT_DIR} = $self->{geno_dir};
+
+  ## read config
+  $Bio::EnsEMBL::Variation::DBSQL::VCFCollectionAdaptor::CONFIG_FILE = $self->{ga_config};
```

Comment:
This isn't OO. It prevents the VCFCollectionAdaptor from doing all the error checking, meaning you have to do it here. What happens if there's no ga_config?